rfc/0009-rancher-integration-charts.md

```diff
@@ -0,0 +1,174 @@
+  * `catalog.cattle.io/certified`
+  * `catalog.cattle.io/certified`
+  * `catalog.cattle.io/experimental`
+  * `catalog.cattle.io/ui-component`
+  * `catalog.cattle.io/display-name`
+  * `catalog.cattle.io/os`
+- Set `catalog.cattle.io/rancher-version` annotation as is required for
+  user-facing charts.
+- Provide `chart/appreadme.md`.
+- Provide logo for the chart by settings `icon` in `Chart.yaml`.
+- Rancher automatically creates a list of container images from charts for
+  airgap. Adopt Rancher's `repository` and `tag` values in values.yml.
+- Provide our own `serviceaccount`.
+- Define k8s min/max version with `kubeVersion` in `Chart.yaml`.
+- Add scheduling options at a pod (affinity) and node level
+  (taints/tolerations/selectors).
+- Provide `annotations.catalog.cattle.io/upstream-version`. This could be
+  automatically bumped on release.
+- Provide `questions.yaml` for defaults on Rancher Explorer UI.
+
+The Kubewarden stack has the following dependencies:
+- Designated main chart: `kubewarden-controller`. This will be the main charts
+  to build a Rancher package from.
+- Hard dependencies: 
+  * CRDs: in `kubewarden-crds` chart. On the main Kubewarden chart
+    (`kubewarden-controller`), set `catalog.cattle.io/auto-install:
+    kubewarden-crds=match` that points to the right CRD version.
+    This annotation accepts `match`, or a hardcoded version (e.g: `1.1.0`)
+    We ought to set it to a hardcoded version, to be able to release patch
+    versions.  of `kubewarden-controller` independently.
+  * `cert-manager`: considered a precondition of Rancher Explorer and out of
+    its scope.
+- Soft (optional) dependencies:
+  `open-telemetry/opentelemetry-operator`
+  `grafana`
+  `kube-prometheus-stack`
+  `jaeger-operator`
+  `kubewarden-defaults`.
+
+  For the ones depending on CRDs (`kubewarden-defaults`,
+    `opentelemetry-operator`, `jaeger-operator`), Rancher charts follow a 3 step
+  process:
+  * Wrap the custom resource in a conditional that is tied to a values.yaml
+    parameter.
+  * Talk to the Rancher Explorer UI team to implement a solution and create a
+  dashboard issue. Make them aware that if a certain CRD is defined we want to
+  flip the new added value to true. If there is configuration available on that
+  resource then it would be grayed out in the UI if the CRD is not available,
+  and editable if the CRD is available.
+  * Document this in our charts.
+  
+  For the ones not depending on CRDs (`grafana`, `prometheus`), we expect end
+  users to deploy/reuse a chart for them and configure the endpoints themselves.
+
+All of the modifications to our codebases will be carried in `rancher-X`
+branches, where `X` is the tag being targeted.
+They will be feature branches with parent being the commit tagged `X`. Using
+these branches allows for easy rebases and cherry-picks, and documenting
+needed changes per commit. 
+
+We expect that all changes will be mergable against main, but having the branch
```

Comment:
```suggestion
We expect that all changes will be mergable against `main`, but having the branch
``

Reply:
done in https://github.com/kubewarden/rfc/pull/11/commits/914babd85855ed860df743e6091cdc093e244493.